pkg/controller/sparkapplication/controller.go

```diff
@@ -639,6 +644,16 @@ func (c *Controller) submitSparkApplication(app *v1beta1.SparkApplication) *v1be
 		return app
 	}
 
+	// Use batch scheduler to perform app submit tasks
```

Comment:
Why not perform the optional batch scheduling before the application gets submitted? This way, we don't need to rely on an update to the `SparkApplicationSpec` to trigger a re-submission to make the batch scheduling effective.

Reply:
If batch scheduling needs to be applied after the application gets submitted, but before the pods get scheduled, the webhook is the right place to apply the batch scheduling settings.

Reply:
Based on the testing, it would be a long process before actual creating the pods, the pods can be patched with the correct annotation. But as you said. there could be a competition. 
what about?
1. Invoking spark batch scheduler before submitting and update the spark app.
or 2. Adding this logic into webhook which doesn't sound a proper place to me.

Reply:
This patch has been updated to the direction of option 1.

Reply:
I like option 2. With option 1, you need to make sure that wherever `submitSparkApplication` gets called, the logic of applying batch scheduling gets added. With option 2, there's no need to change the `SparkApplicationSpec`, and the webhook is a central place for applying batch scheduling mutations directly to pods.

Reply:
@liyinan926  thanks, patch updated heading to the option 2